ostd/src/mm/kspace/kva.rs

```diff
@@ -0,0 +1,362 @@
+    pub fn new(size: usize) -> Self {
+        let allocator = M::select_allocator();
+        let range = allocator.alloc(size).unwrap();
+        let num_cpus_ = num_cpus();
+        Self {
+            range,
+            num_cpus_,
+            phantom: PhantomData,
+        }
+    }
+
+    pub fn start(&self) -> Vaddr {
+        self.range.start
+    }
+
+    pub fn end(&self) -> Vaddr {
+        self.range.end
+    }
+
+    pub fn range(&self) -> Range<Vaddr> {
+        self.range.start..self.range.end
+    }
+
+    /// Map pages into the kernel virtual area.
+    /// # Safety
+    /// The caller should ensure either the mapped pages or the range to be used doesn't
+    /// violate the memory safety of kernel objects.
+    pub unsafe fn map_pages<T: PageMeta>(
+        &mut self,
+        range: Range<Vaddr>,
+        pages: impl Iterator<Item = Page<T>>,
+    ) {
+        assert!(self.start() <= range.start && self.end() >= range.end);
+        let page_table = KERNEL_PAGE_TABLE.get().unwrap();
+        let prop = PageProperty {
+            flags: PageFlags::RW,
+            cache: CachePolicy::Writeback,
+            priv_flags: PrivilegedPageFlags::GLOBAL,
+        };
+        let mut cursor = page_table.cursor_mut(&range).unwrap();
+        let flusher = TlbFlusher::new(CpuSet::new_full(), disable_preempt());
+        let mut va = self.start();
+        for page in pages.into_iter() {
+            if let Some(old) = cursor.map(page.into(), prop) {
+                flusher.issue_tlb_flush_with(TlbFlushOp::Address(va), old);
+                flusher.dispatch_tlb_flush();
+            }
+            va += PAGE_SIZE;
+        }
+        tlb_flush_addr_range(&range);
+    }
+
+    /// This function returns the page usage type based on the provided virtual address `addr`.
+    /// This function returns None in the following cases:
+    /// * If the address is not mapped (`NotMapped`).
+    /// * If the address is mapped to a `MappedUntracked` page.
+    pub fn get_page_type(&self, addr: Vaddr) -> Option<PageUsage> {
+        assert!(self.start() <= addr && self.end() >= addr);
+        let start = addr.align_down(PAGE_SIZE);
+        let vaddr = start..start + PAGE_SIZE;
+        let page_table = KERNEL_PAGE_TABLE.get().unwrap();
+        let mut cursor = page_table.cursor(&vaddr).unwrap();
+        let query_result = cursor.query().unwrap();
+        match query_result {
+            PageTableItem::Mapped {
+                va: _,
+                page,
+                prop: _,
+            } => Some(page.usage()),
+            _ => None,
+        }
+    }
+
+    /// Get the mapped page.
+    /// This function returns None in the following cases:
+    /// * if the provided page type doesn't match the actual mapped one.
+    /// * If the address is not mapped (`NotMapped`).
+    /// * If the address is mapped to a `MappedUntracked` page.
+    pub fn get_page<T: PageMeta>(&self, addr: Vaddr) -> Option<Page<T>> {
```

Comment:
Following the above comment, the signature of this function should be `get_page(&self, addr: Vaddr) -> Option<DynPage>`. This function should only be implemented for `Kva<Tracked>`.

For `Kva<Untracked>`, it should be `get__untracked_page(&self, addr: Vaddr) -> Option<(Paddr, usize)>`.